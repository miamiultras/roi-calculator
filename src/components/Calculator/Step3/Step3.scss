.step3-container {
  width: 100%;
  color: #fff;
  h1 {
    font-family: 'Roboto';
    font-weight: 300;
    color: #fff;
    font-size: 36px;
    margin: 20px 0 10px 0;
    text-transform: uppercase;
  }

  img {
    margin-top: 34px;
    height: 76px;
  }

  .currency-input {
    position: relative;
    width: 242px;
    margin: 0 auto;
    input[type="text"] {
      width: 242px;
      padding: 4px 8px;
      text-align: right;
      color: #999;
      font-size: 21px;
      margin-bottom: 50px;
      padding-right: 20px;
    }

    .currency-sign {
      font-size: 19px;
      color: #777;
      position: absolute;
      top: 6px;
      right: 7px;
    }
  }
}
