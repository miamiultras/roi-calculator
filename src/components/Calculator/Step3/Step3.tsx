import React from 'react';
import './Step3.scss';

import icon from '../../../assets/images/tickets.svg';
import Paragraph from '../../Paragraph/Paragraph';
import NextButton from '../../NextButton/NextButton';
import BackButton from '../../BackButton/BackButton';
import CalculatorNavigation from '../../CalculatorNavigation/CalculatorNavigation';

const Step3 = (
  { pricePerTicket,
    handleChange,
    handleSubmit,
  }: {
      pricePerTicket: string,
      handleChange: any,
      handleSubmit: any,
    }) => (
    <div className='step3-container'>
      <div className='title'>
        <h1>Live PPV Event ROI Calculator</h1>
      </div>
      <Paragraph>
        Working with a Live PPV specialist can boost your conversion rate and ultimately your ROI.
    </Paragraph>
      <img src={icon} alt='icon' />
      <Paragraph>
        How much do you plan to charge per ticket for your Live PPV event?
    </Paragraph>

      <div className='currency-input'>
        <input
          id='pricePerTicket'
          type='text'
          name='pricePerTicket'
          value={pricePerTicket}
          onChange={handleChange}
        />
        <span className='currency-sign'>&euro;</span>
      </div>
      <NextButton name='see the results' to='step3' action={handleSubmit} />
      <BackButton name='back' to='step2' />
      <CalculatorNavigation step={3} />
    </div>
  );

export default Step3;
