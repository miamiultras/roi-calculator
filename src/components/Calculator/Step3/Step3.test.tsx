import * as React from 'react';
import { shallow } from 'enzyme';
import Step3 from './Step3';

function functionMock() { }

function setup() {
  const props = {};
  const wrapper = shallow(
    <Step3
      pricePerTicket='19.99'
      handleChange={functionMock}
      handleSubmit={functionMock}
    />
  );
  return {
    wrapper,
    props,
  };
}

describe('Step3 component', () => {
  it('should match snapshot if no change was intended', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
