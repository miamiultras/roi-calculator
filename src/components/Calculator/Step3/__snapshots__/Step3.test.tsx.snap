// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step3 component should match snapshot if no change was intended 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Step3
    handleChange={[Function]}
    handleSubmit={[Function]}
    pricePerTicket="19.99"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title"
        >
          <h1>
            Live PPV Event ROI Calculator
          </h1>
        </div>,
        <Paragraph>
          Working with a Live PPV specialist can boost your conversion rate and ultimately your ROI.
        </Paragraph>,
        <img
          alt="icon"
          src="tickets.svg"
        />,
        <Paragraph>
          How much do you plan to charge per ticket for your Live PPV event?
        </Paragraph>,
        <div
          className="currency-input"
        >
          <input
            id="pricePerTicket"
            name="pricePerTicket"
            onChange={[Function]}
            type="text"
            value="19.99"
          />
          <span
            className="currency-sign"
          >
            €
          </span>
        </div>,
        <NextButton
          action={[Function]}
          name="see the results"
          to="step3"
        />,
        <BackButton
          name="back"
          to="step2"
        />,
        <CalculatorNavigation
          step={3}
        />,
      ],
      "className": "step3-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Live PPV Event ROI Calculator
          </h1>,
          "className": "title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Live PPV Event ROI Calculator",
          },
          "ref": null,
          "rendered": "Live PPV Event ROI Calculator",
          "type": "h1",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Working with a Live PPV specialist can boost your conversion rate and ultimately your ROI.",
        },
        "ref": null,
        "rendered": "Working with a Live PPV specialist can boost your conversion rate and ultimately your ROI.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "icon",
          "src": "tickets.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "How much do you plan to charge per ticket for your Live PPV event?",
        },
        "ref": null,
        "rendered": "How much do you plan to charge per ticket for your Live PPV event?",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              id="pricePerTicket"
              name="pricePerTicket"
              onChange={[Function]}
              type="text"
              value="19.99"
            />,
            <span
              className="currency-sign"
            >
              €
            </span>,
          ],
          "className": "currency-input",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "pricePerTicket",
              "name": "pricePerTicket",
              "onChange": [Function],
              "type": "text",
              "value": "19.99",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "€",
              "className": "currency-sign",
            },
            "ref": null,
            "rendered": "€",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "action": [Function],
          "name": "see the results",
          "to": "step3",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "back",
          "to": "step2",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "step": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
          >
            <h1>
              Live PPV Event ROI Calculator
            </h1>
          </div>,
          <Paragraph>
            Working with a Live PPV specialist can boost your conversion rate and ultimately your ROI.
          </Paragraph>,
          <img
            alt="icon"
            src="tickets.svg"
          />,
          <Paragraph>
            How much do you plan to charge per ticket for your Live PPV event?
          </Paragraph>,
          <div
            className="currency-input"
          >
            <input
              id="pricePerTicket"
              name="pricePerTicket"
              onChange={[Function]}
              type="text"
              value="19.99"
            />
            <span
              className="currency-sign"
            >
              €
            </span>
          </div>,
          <NextButton
            action={[Function]}
            name="see the results"
            to="step3"
          />,
          <BackButton
            name="back"
            to="step2"
          />,
          <CalculatorNavigation
            step={3}
          />,
        ],
        "className": "step3-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Live PPV Event ROI Calculator
            </h1>,
            "className": "title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Live PPV Event ROI Calculator",
            },
            "ref": null,
            "rendered": "Live PPV Event ROI Calculator",
            "type": "h1",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Working with a Live PPV specialist can boost your conversion rate and ultimately your ROI.",
          },
          "ref": null,
          "rendered": "Working with a Live PPV specialist can boost your conversion rate and ultimately your ROI.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "icon",
            "src": "tickets.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "How much do you plan to charge per ticket for your Live PPV event?",
          },
          "ref": null,
          "rendered": "How much do you plan to charge per ticket for your Live PPV event?",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                id="pricePerTicket"
                name="pricePerTicket"
                onChange={[Function]}
                type="text"
                value="19.99"
              />,
              <span
                className="currency-sign"
              >
                €
              </span>,
            ],
            "className": "currency-input",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "pricePerTicket",
                "name": "pricePerTicket",
                "onChange": [Function],
                "type": "text",
                "value": "19.99",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "€",
                "className": "currency-sign",
              },
              "ref": null,
              "rendered": "€",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "action": [Function],
            "name": "see the results",
            "to": "step3",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "back",
            "to": "step2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "step": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
